src/XRay/XRayOptions.ts

```diff
@@ -0,0 +1,258 @@
+ *
+ *  !!!!!!! SOURCE GETS TRANSPILED BY TYPESCRIPT. EDIT TS FILE ONLY. !!!!!!!
+ *
+ *  Authors:
+ *  - Sophie Bremer
+ *
+ * */
+
+
+'use strict';
+
+
+/* *
+ *
+ *  Imports
+ *
+ * */
+
+
+import type {
+    MorningstarConverterOptions,
+    MorningstarHoldingAmountOptions,
+    MorningstarHoldingWeightOptions,
+    MorningstarOptions
+} from '../Shared/MorningstarOptions';
+
+
+/* *
+ *
+ *  API Options
+ *
+ * */
+
+
+export interface XRayBenchmarkDataPointOptions {
+
+    /**
+     * A comma-separated list of benchmark data points to return.
+     */
+    dataPoints: Array<XRayBenchmarkDataPoints>;
+
+    /**
+     * Type of data points.
+     */
+    type: 'benchmark';
+
+}
+
+
+export type XRayBenchmarkDataPoints = (
+    | 'GrowthSeries'
+    | 'HistoricalPerformanceSeries'
+    | 'PerformanceReturn'
+    | ['PerformanceReturn', ...Array<string>]
+    | 'ShowBreakdown'
+);
+
+
+export interface XRayConverterOptions extends MorningstarConverterOptions {
+
+}
+
+
+export type XRayDataPointOptions = (
+    | XRayBenchmarkDataPointOptions
+    | XRayHoldingDataPointOptions
+    | XRayPortfolioDataPointOptions
+);
+
+
+export interface XRayHoldingDataPointOptions {
+
+    /**
+     * A comma-separated list of holding data points to return.
+     */
+    dataPoints: Array<XRayHoldingDataPoints>;
+
+    /**
+     * Type of data points.
+     */
+    type: 'holding';
+
+}
+
+
+/**
+ * @todo Consider sub-collections or reduction to `string`
+ */
+export type XRayHoldingDataPoints = (
+    | 'Alpha'
+    | 'ArithmeticMean'
+    | 'AveragePrice'
+    | 'Beta'
+    | 'CalculatedSRRI'
+    | 'CalmarRatio'
+    | 'CouponFrequency'
+    | 'CouponType'
+    | 'CurrencyId'
+    | 'CurrencyName'
+    | 'DiscountPremium'
+    | 'EquityWeighting'
+    | 'EquityWeightingLong'
+    | 'HoldingId'
+    | 'HoldingType'
+    | 'InitialAmount'
+    | 'InitialWeight'
+    | 'Kurtosis'
+    | 'MarketValue'
+    | 'MorningstarRiskM255'
+    | 'Name'
+    | 'NAVCFLC'
+    | 'NAVCFLE'
+    | 'NAVCFLP'
+    | 'NAVEPLC'
+    | 'NAVEPLE'
+    | 'NAVEPLP'
+    | 'OngoingCharge'
+    | 'PortfolioId'
+    | 'RiskRewardVolatility'
+    | 'SecurityId'
+    | 'SecurityType'
+    | 'SharpeRatio'
+    | 'SortinoRatio'
+    | 'SRRI'
+    | 'StandardDeviation'
+    | 'StarRating'
+    | 'TDYNAV'
+    | 'UCITS'
+    | 'Weight'
+);
+
+
+export interface XRayOptions extends MorningstarOptions {
+
+    /**
+     * Morningstar-defined unique identifier of a benchmark.
+     */
+    benchmarkId?: string;
+
+    /**
+     * Options related to the x-ray data parser.
+     */
+    converter?: XRayConverterOptions;
+
+    /**
+     * Currency of the portfolio. ISO 4217 3-character currency code.
+     */
+    currencyId?: string;
+
+    /**
+     * Data points for the x-ray.
+     */
+    dataPoints?: XRayDataPointOptions;
+
+    /**
+     * Array of portfolio holdings.
+     */
+    holdings?: (Array<MorningstarHoldingAmountOptions>|Array<MorningstarHoldingWeightOptions>);
```

Comment:
Very long, and written out in full multiple places. I would consider making a type alias for this.